### PR TITLE
Create stubs for MSBuild tasks

### PR DESCRIPTION
**Pull request by smadala**
## Description
- Add stubs required to actual code.
- Enable CI
## Related issue
#11 #9 #8

### CODE REVIEW

File: src/Coverage.Report.Tests/Coverage.Report.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (codito) [inline src/Coverage.Report.Tests/Coverage.Report.Tests.csproj]: Should we add a `.props` with versioning and common properties in this PR?

Comment (codito) [reply src/Coverage.Report.Tests/Coverage.Report.Tests.csproj]: Consider including stylecop checks.

File: src/Coverage.Report/Friends.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.CompilerServices;
+
+#region Test Assemblies
+
+[assembly: InternalsVisibleTo("Microsoft.TestPlatform.Build.UnitTests")]
```
Comment (codito) [inline src/Coverage.Report/Friends.cs]: Change project name.

File: src/Coverage.Report/Interfaces/ILcovConvertForwardingApp.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿namespace Coverage.Report.Interfaces
+{
+    using System.Collections.Generic;
+
+    public interface ILcovConvertForwardingApp
```
Comment (codito) [inline src/Coverage.Report/Interfaces/ILcovConvertForwardingApp.cs]: Should we document the interfaces?

Comment (codito) [reply src/Coverage.Report/Interfaces/ILcovConvertForwardingApp.cs]: What does the forwarding app do?

File: src/Coverage.Report/CLIReportTask.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        }
+
+        public CLIReportTask(Trace trace)
+        {
+            this.trace = trace;
+        }
+
+        public override bool Execute()
+        {
+            this.trace.Log("=================================================" + Environment.NewLine +
```
Comment (codito) [inline src/Coverage.Report/CLIReportTask.cs]: Is this a console output or a verbose information? `Log` may need verbosity enum?

File: src/Coverage.Report.Tests/ConvertToLcovTaskTests.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
```
Comment (codito) [inline src/Coverage.Report.Tests/ConvertToLcovTaskTests.cs]: Intentional to keep test code in `src/` directory? Would it not clutter structure when we start adding test assets?

File: src/Coverage.Report.Tests/Coverage.Report.Tests.csproj (hunk 1 above)
Comment (smadala) [reply src/Coverage.Report.Tests/Coverage.Report.Tests.csproj]: >Should we add a .props with versioning and common properties in this PR?

We can do it in PR in which we add second test project.

>Consider including stylecop checks.

May be in this PR or immediate next PR.

File: src/Coverage.Report/Friends.cs (hunk 2 above)
Comment (smadala) [reply src/Coverage.Report/Friends.cs]: Good catch. Thanks

File: src/Coverage.Report/Interfaces/ILcovConvertForwardingApp.cs (hunk 3 above)
Comment (smadala) [reply src/Coverage.Report/Interfaces/ILcovConvertForwardingApp.cs]: > Should we document the interfaces?

Yes, Will do in this PR or next PR. I want to unblock the contributes ASAP.

>What does the forwarding app do?

Didn't think much about it. It is copied from [VSTestForwardingApp](https://github.com/Microsoft/vstest/blob/master/src/Microsoft.TestPlatform.Build/Tasks/VSTestForwardingApp.cs). I'm thinking Tasks classes just have responsibility to get parameter from target. The core logic should happen in forwarding app(or any better name we can give).

File: src/Coverage.Report.Tests/ConvertToLcovTaskTests.cs (hunk 5 above)
Comment (smadala) [reply src/Coverage.Report.Tests/ConvertToLcovTaskTests.cs]: Yes, It is intentional. Copied from https://github.com/dotnet/corefx/tree/master/src/System.Buffers and some other great repos. TestAssets can have parallel to `src` dir?

Comment (smadala) [reply src/Coverage.Report.Tests/ConvertToLcovTaskTests.cs]: Just now one good thing I observed having tests in `test` directory is all the product code changes appears first while reviewing the code. Let's think about it. I'm open to both the options.

**Pull request merged**